test/e2e/case5_policy_automation_test.go

```diff
@@ -137,9 +142,29 @@ var _ = Describe("Test policy automation", func() {
 				Expect(err).To(BeNil())
 				_, err = utils.KubectlWithOutput("get", "ansiblejobs", "-n", testNamespace)
 				Expect(err).Should(BeNil())
+				spec := ansiblejobList.Items[0].Object["spec"]
+				extraVars := spec.(map[string]interface{})["extra_vars"].(map[string]interface{})
+
+				return len(extraVars)
+			}, 30, 1).Should(Equal(9))
+
+			By("Check each violation context field in extraVars")
+			ansiblejobList, err := clientHubDynamic.Resource(gvrAnsibleJob).Namespace(testNamespace).List(
+				context.TODO(), metav1.ListOptions{},
+			)
+			Expect(err).To(BeNil())
+			spec := ansiblejobList.Items[0].Object["spec"]
+			extraVars := spec.(map[string]interface{})["extra_vars"].(map[string]interface{})
+			Expect(extraVars["policy_name"]).To(Equal("case5-test-policy"))
+			Expect(extraVars["namespace"]).To(Equal("policy-propagator-test"))
+			Expect(len(extraVars["target_clusters"].([]interface{}))).To(Equal(1))
+			Expect(extraVars["target_clusters"].([]interface{})[0]).To(Equal("managed1"))
+			Expect(len(extraVars["policy_set"].([]interface{}))).To(Equal(1))
+			Expect(extraVars["policy_set"].([]interface{})[0]).To(Equal("case5-test-policyset"))
+			managed1 := extraVars["policy_violation_context"].(map[string]interface{})["managed1"]
+			compliant := managed1.(map[string]interface{})["compliant"]
+			Expect(compliant).To(Equal(string(policiesv1.NonCompliant)))
```

Comment:
Is the hub cluster and the message from the violation context getting checked too?

Reply:
Good question. 

1) I found `message` is from `PolicyStatus.Details` which is used by the replicated policy so it will always empty when checking the root policy, so maybe we should change the design and move the `message` into `PolicyViolationContext` for each replicated policies?

2) I use policy.GetClusterName() to get the root policy hub cluster name and I think it doesn't work since there isn't a cluster name in the metadata. I'm looking for a better way to get the hub cluster name.